### PR TITLE
Use spotless instead of kotlinter

### PR DESCRIPTION
**Pull request by StefanBratanov**
Use spotless with ktlint (same as kotlinter), but it will also allow linting the java classes.

https://github.com/diffplug/spotless/tree/main/plugin-gradle

Required updating ktlint from version `45.2` used by kotlinter to `0.50`.

New rules were added for ktlint which required most of the changes in this PR after doing `spotlessApply`

- https://pinterest.github.io/ktlint/0.50.0/rules/standard/#classobject-naming

Renamed couple of things because of class-naming violation:

- KEY_TYPE enum in `Key.kt` to KeyType
- UNINITIALIZED_VALUE in `Delegates.kt` to UninitializedValue

Java classes were updated with the `googleJavaFormat` formatting after applying `spotlessApply`

Disabled enforcing of `trailing comma`

### CODE REVIEW

**Comment by Nashatyrev** [comment]
Do we really want to _enforce_ trailing comma?

**Comment by StefanBratanov** [comment]
> Do we really want to _enforce_ trailing comma?

We can disable it yeah.

**Comment by StefanBratanov** [comment]
> > Do we really want to _enforce_ trailing comma?
> 
> We can disable it yeah.

Done. Much smaller changed files. :)

**Review by Nashatyrev — APPROVED** [review]
LGTM

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1 @@
+* text eol=lf
```
[the remark this answers is not in the document]
Comment (Nashatyrev) [reply .gitattributes:1]: Created PR on this issue: https://github.com/libp2p/jvm-libp2p/pull/335

[the remark this answers is not in the document]
Comment (StefanBratanov) [reply .gitattributes:1]: Thanks for the fix. I think it was spotless complaining about line endings and saw discussion to add this to .gitattributes

**Pull request merged**